gnovm/pkg/gnolang/helpers.go

```diff
@@ -21,6 +22,7 @@ var ReGnoRunPath = regexp.MustCompile(`^gno\.land/r/g[a-z0-9]+/run$`)
 // IsRealmPath determines whether the given pkgpath is for a realm, and as such
 // should persist the global state.
 func IsRealmPath(pkgPath string) bool {
+	pkgPath = filepath.Clean(pkgPath)
```

Comment:
we can't use a platform-specific call in a helper used in consensus
furthermore, transforming the path is a bad idea, this should actually reject relative paths IMO

Reply:
I get it, indeed i will pass the PR in draft until i can apply in a good way the pattern Morgan suggested without breaking test & linters, thanks for comment

Reply:
good catch, this should use `path`

Reply:
I don't think it should use `path`, to me this PR reveal bad usages of `IsRealmPath` with filesystem paths.

This works in the very narrow case of the examples dir because it has the same structure as the pkgpaths but in most cases a filesystem path cannot be converted to a pkg path this way.

A simple "hotfix" could be to call `filepath.Clean` at the bad call sites of `IsRealmPath`

I think a more durable fix would be to declare a `type PkgPath string` and `IsRealmPath(pkgPath PkgPath) to better differentiate pkgpaths and filesystem paths but it's out of scope of this

Reply:
@n0izn0iz `path` is os agnostic and always uses `/` as a dir separator

in any case we shouldn't "clean" here as there's nothing to clean to begin with

Reply:
I reverted to previous method with gno.land/p prefix, but indeed the problem was coming from the gno linter that fill the memPkgPaths with path from filesystem